### PR TITLE
Topic2

### PR DESCRIPTION
**Pull request by LucasAguirre**
Topic 2 exercises

### CODE REVIEW

File: java-bootcamp-2018-tdd/src/test/java/com/globant/java_bootcamp_2018_tdd/RecentFileListTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import org.junit.Test;
+
+public class RecentFileListTest {
+
+	// For this tests the list of recent opened files have a size of 5, the
+	// predetermined(in the recent file list class) is 15
+	RecentFileList list = new RecentFileList(5);
+
+	@Test
+	public final void whenTheProgramIsRunForTheFirstTimeTheListIsEmpty() {
```
Comment (salaesp) [inline java-bootcamp-2018-tdd/src/test/java/com/globant/java_bootcamp_2018_tdd/RecentFileListTest.java]: add test to the start of the method. Also you can use a shorter name and add the full description as comments

File: java-bootcamp-2018-tdd/src/test/java/com/globant/java_bootcamp_2018_tdd/RecentFileListTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	@Test
+	public final void ifAnOpenedFileAlreadyExistsInTheRecentFileListItIsBumpedToTheTopNotDuplicatedInTheList() {
+		File file1 = new File("t3-1");
+		File file2 = new File("t3-2");
+		File file3 = new File("t3-3");
+		list.openFile(file1);
+		list.openFile(file2);
+		list.openFile(file3);
+		list.openFile(file1);
+		assertEquals("A reopen file is not on the top of the list", file1, list.lastElemment());
```
Comment (salaesp) [inline java-bootcamp-2018-tdd/src/test/java/com/globant/java_bootcamp_2018_tdd/RecentFileListTest.java]: you should check that the file is not duplicated

File: java-bootcamp-2018-tdd/src/test/java/com/globant/java_bootcamp_2018_tdd/RecentFileListTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		File file1 = new File("t4-1");
+		File file2 = new File("t4-2");
+		File file3 = new File("t4-3");
+		File file4 = new File("t4-4");
+		File file5 = new File("t4-5");
+		File file6 = new File("t4-6");
+		list.openFile(file1);
+		list.openFile(file2);
+		list.openFile(file3);
+		list.openFile(file4);
```
Comment (salaesp) [inline java-bootcamp-2018-tdd/src/test/java/com/globant/java_bootcamp_2018_tdd/RecentFileListTest.java:59]: I would check if every file is in the list, just not the first one. Also check the size

File: java-bootcamp-2018-tdd/src/test/java/com/globant/java_bootcamp_2018_tdd/RecentFileListTest.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		list.openFile(file5);
+		list.openFile(file6);
+		assert (!list.exist(file1));
+		assert (list.exist(file2));
+	}
+
+	@Test
+	public final void testDeleteAEntry() {
+		File file = new File("t5-1");
+		list.openFile(file);
```
Comment (salaesp) [inline java-bootcamp-2018-tdd/src/test/java/com/globant/java_bootcamp_2018_tdd/RecentFileListTest.java]: I would also add some tests for removing an element from a not empty list

File: java-bootcamp-2018-tdd/src/main/java/com/globant/java_bootcamp_2018_tdd/RecentFileList.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+		this.maxSize = size;
+	}
+
+	public int size() {
+		return list.size();
+	}
+
+	public void openFile(File file) {
+		if (list.size() >= maxSize) {
+			list.remove(0);
```
Comment (salaesp) [inline java-bootcamp-2018-tdd/src/main/java/com/globant/java_bootcamp_2018_tdd/RecentFileList.java]: never use lists indexes. You can use the iterator instead.

File: java-bootcamp-2018-tdd/src/main/java/com/globant/java_bootcamp_2018_tdd/RecentFileList.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+		}
+		if (!list.contains(file)) {
+			list.add(file);
+		} else {
+			list.remove(file);
```
Comment (salaesp) [inline java-bootcamp-2018-tdd/src/main/java/com/globant/java_bootcamp_2018_tdd/RecentFileList.java:35]: you could use the public method on this class to delete the file so you have the whole functionality into one algorithm

**Review by salaesp — CHANGES_REQUESTED** [review]
Please recheck the tests cases, there should be more taken into account.

File: java-bootcamp-2018-tdd/src/main/java/com/globant/java_bootcamp_2018_tdd/App.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package com.globant.java_bootcamp_2018_tdd;
+
+/**
+ * Hello world!
```
Comment (salaesp) [inline java-bootcamp-2018-tdd/src/main/java/com/globant/java_bootcamp_2018_tdd/App.java]: You are missing the blog part

File: java-bootcamp-2018-tdd/src/main/java/com/globant/blog/Blog.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
+package com.globant.blog;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Blog {
```
Comment (sebastian-garofalo) [inline java-bootcamp-2018-tdd/src/main/java/com/globant/blog/Blog.java:6]: Lets start using javadoc

File: java-bootcamp-2018-tdd/src/main/java/com/globant/blog/Blog.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+	public void addEntry(Post post) {
+		this.posts.add(post);
+	}
+	public int quantityPost() {
+		return this.posts.size();
+	}
+
+	public void deleteEntry(Post post) {
+	/*	for(Post it:posts) {
```
Comment (sebastian-garofalo) [inline java-bootcamp-2018-tdd/src/main/java/com/globant/blog/Blog.java]: please remove unused code
